File: STRSection.m
classdef STRSection
    properties
        Id;
        Name;
        Ax;
        Ix;  %polar MI (torsion)
        Iy;  %second MI
        Iz;  %MI around vertical axis

    end

    methods
        function obj = STRSection(id,name,ax,ix,iy,iz)
            obj.Id = id;
            obj.Name = name;
            obj.Ax = ax;
            obj.Ix = ix;
            obj.Iy = iy;
            obj.Iz = iz;
        end

        function ToString(obj)
            fprintf('Section (%s) %i\n',obj.Name, obj.Id);
        end
    end
end

File: src/STRSection.cpp
#include "STRSection.hpp"
#include <iostream>

STRSection::STRSection(int id, std::string& name, double ax, double ix, double iy, double iz)
	: Id(id), Name(name), Ax(ax), Ix(ix), Iy(iy), Iz(iz) {
};

void STRSection::ToString() const {
	std::cout << "Section (" << Name << ") #" << Id << "\n";
};